test-data/unit/pythoneval.test

```diff
@@ -321,8 +325,8 @@ reveal_type(p.open(errors='replace', mode='rb'))
 mode = 'rb'
 reveal_type(p.open(mode=mode, errors='replace'))
 [out]
-_program.py:3: note: Revealed type is 'typing.BinaryIO'
-_program.py:4: note: Revealed type is 'typing.BinaryIO'
+_program.py:3: note: Revealed type is 'typing.IO[Any]'
```

Comment:
These seem like regressions. Why can mypy no longer infer BinaryIO here?

Reply:
`p.open(mode='rb', errors='replace')` raises an exception (`ValueError: binary mode doesn't take an errors argument`), so its type is not really `BinaryIO`.

`mypy` infers it is a `IO[Any]` because of the [fallback overload](https://www.python.org/dev/peps/pep-0586/#interactions-with-overloads) with `mode: str`, but ideally `open` should have a return type of `NoReturn` in this case. I wonder if adding an overload variant with a return type of `NoReturn` would solve the problem.

Reply:
`NoReturn` is not really intended for error cases. Falling back to the default return type on error case seems fine to me.

Reply:
To be clear, when you say "default return type", do you mean `IO[Any]` or `BinaryIO`?

Reply:
Looking at this it seems to me it is better to improve this plugin instead of removing it. For example, the `p.open(mode='rb', errors='replace')` call will silently fall back to the default overload, but with the plugin we can detect the error and give a nice error message (matching the runtime one).

Reply:
I modified the plugin to detect use of text-only arguments in binary mode.